Core/include/Acts/TrackFitting/Chi2Fitter.hpp

```diff
@@ -352,13 +349,28 @@ class Chi2Fitter {
 
         // add a full TrackState entry multi trajectory
         // (this allocates storage for all components, we will set them later)
-        result.lastTrackIndex = result.fittedStates->addTrackState(
-            ~(TrackStatePropMask::Smoothed | TrackStatePropMask::Filtered),
-            result.lastTrackIndex);
+
+        size_t currentTrackIndex = SIZE_MAX;
+        if (updateNumber == 0) {
```

Comment:
I think we should remove this `updateNumber` check in favor of doing first the `visitBackwards` and then if we did not have success, doing the `addTrackState`.

Reply:
As @benjaminhuth and I just found out: We need a check in some place because we can't `visitBackwards` in the very first step, since `lastTrackIndex = kInvalid` let's this crash.